src/main/java/bong/utils/Parser.java

```diff
@@ -0,0 +1,70 @@
+package bong.utils;
+
+import bong.command.*;
+
+import bong.task.Deadline;
+import bong.task.Event;
+import bong.task.Todo;
+
+/**
+ * The {@code Parser} class is responsible for parsing user input into commands
+ * that the Bong application can execute.
+ */
+public class Parser {
+    /**
+     * Parses the user input and returns the corresponding {@code Command} object.
+     *
+     * @param userInput The input string provided by the user.
+     * @return The {@code Command} object representing the user's command.
+     * @throws BongException If the user input cannot be parsed into a valid command.
+     */
+    public Command parseCommand(String userInput) throws BongException {
+        String[] parts = userInput.split(" ", 2);
+        String commandWord = parts[0].toLowerCase();
+        String arguments = parts.length > 1 ? parts[1] : "";
+
+        switch (commandWord) {
+            case "bye" -> {
```

Comment:
Perhaps the indentation of `case` clause could be updated according to the Java coding standard?
Noticed that the same occurs in `Storage.java`

Reply:
Adding on to what Gilbb has mentioned, it seems that the convention for the switch arrow-notation has been agreed to be as such according to this [issue](https://github.com/nus-cs2103-AY2425S1/forum/issues/49):

```
switch (data.charAt(0)) {
case 'T' -> functionT();
case 'E' -> functionE();
case 'D' -> functionD();
default -> functionDef();
}; 
```
Also, this is not a violation but perhaps the `default` case could be wrapped with `{ }` to standardize it? 😄